src/app/reporting/ReportScheduler.h

```diff
@@ -189,10 +189,17 @@ class ReportScheduler : public ReadHandler::Observer, public ICDStateObserver
         BitFlags<ReadHandlerNodeFlags> mFlags;
     };
 
+    ReportScheduler() : mTimerDelegate(nullptr) {}
     ReportScheduler(TimerDelegate * aTimerDelegate) : mTimerDelegate(aTimerDelegate) {}
 
     virtual ~ReportScheduler() = default;
 
+    void Init(TimerDelegate * aTimerDelegate)
```

Comment:
How is this method supposed to be used with this constructor:
`ReportScheduler(TimerDelegate * aTimerDelegate) : mTimerDelegate(aTimerDelegate) {}`?

It would overwrite the TimerDelegate previously used, which defeats the purpose of having a constructor that accepts a TimerDelegate.

Reply:
Yes, right, Global<T> doesn't do anything except default constructors, so I added this Init method instead. I didn't delete the constructor with the pointer as I suspect it's already used by people. The intention was you can use the default constructor and then Init() later, or you can use the constructor with a pointer, in which case you don't need to Init().

Reply:
I think the better pattern is to have the Global be of a sub-class or wrapper struct with the default constructor we want.